src/js/utils/strings.ts

```diff
@@ -16,7 +16,7 @@ export function pad(str, length, padder) {
 }
 
 // Get the value of a case-insensitive attribute in an XML node
-export function xmlAttribute(xml, attribute) {
+export function xmlAttribute(xml: Element, attribute: string): string {
```

Comment:
what is the difference between `Element` for the xml here and `Document` above?

Reply:
Relooking at this, it looks like above should be returning an `XMLDocument` instead. This `Element` is just an XML node. Apparently `Element` inherits from `Node`, `ChildNode`, and `ParentNode`.
As far as I know the `XMLDocument` can contain these elements.